# MUHAMMAD SIDDIQ 19B-071-SE 

# Python User Define Function Practice Activity - 2 

Asst. Prof. Syed Faisal Ali              $\;\;\;\;\;\;$              Programming Fundamentals - FALL 2019 $\;\;\;\;\;\;$ Software Engineering                     $\;\;\;\;\;\;$                   Dated: 27 Nov 2019

Question 1: Create a function to find the following:
If the base of triangle is 3 cm long and its equilateral triangle and the radius of circle is 1.5 cm then find the area of triangle shaded. 


![proj](q1.jpg)

In [39]:
def area(b,r):
    h=b*r/2
    return h
area(3,1.5)

2.25

Question 2: Create a function which can read a dictionary of your family members such as 5 members. 
1 Abbu, 1 Ammi, 2 Brothers 1 Sister. Now feed this data in dictionary in terms of name and relations.
The UDF will ask findrelation() in this you will enter Brother it will return the names of two brothers you have inserted. In case if the relation is not found it will return “Sorry the relation doesn’t exist in your family.” 


In [40]:
def findrelation(d):
    dict={'Mother': 'Huma', 'Father': 'Rafiq', 'Brother': ('Ali','Raffay'),'Sister':'Noor'}
    a=dict[d]
    return a
findrelation('Brother')
    

('Ali', 'Raffay')

Question 3: Create a function to find the following:
If the base of triangle is 5 cm long and its equilateral triangle and the radius of circle is 2.25 cm then find the area of triangle shaded. 


![proj](q3.jpg)

In [44]:
import math
def f(b):
    h=b**2*((math.sqrt(3))/4)
    return h
f(5)

10.825317547305483

Question 4:
Create a function that takes a list of random numbers from users and add only those which are even. If all the numbers are odd it will return sorry no even number found.


In [27]:
NumList = []
Even_Sum = 0
Number = int(input("Please enter the Total Number of List Elements: "))
for i in range(1, Number + 1):
    value = int(input("Please enter the Value of %d Element : " %i))
    NumList.append(value)
for j in range(Number):
    if(NumList[j] % 2 == 0):
        Even_Sum = Even_Sum + NumList[j]
    else:
        print("sorry no even number.")
print("\nThe Sum of Even Numbers in this List =  ", Even_Sum)

Please enter the Total Number of List Elements: 4
Please enter the Value of 1 Element : 2
Please enter the Value of 2 Element : 4
Please enter the Value of 3 Element : 6
Please enter the Value of 4 Element : 8

The Sum of Even Numbers in this List =   20


Question 5:
Write a function which can take a list of numbers and it will return sorted list.


In [ ]:
lst=[]
n = int(input("Enter number of elements : "))
for i in range(0,n):
    n=int(input())
    lst.sort()
print(lst.sort)

Question 6:
Write a function that will take the radius and return the perimeter and area of circle with 5% increment.


In [34]:
def findArea(r): 
    PI = 3.142
    return PI * (r*r) * 1.5
print(findArea(5))



117.82499999999999


Question 7:
Write a function that will take the strings as argument and return number of vowels and consonants.


In [38]:
def Check(string, vowels, consonants): 
    final = [each for each in string if each in vowels] 
    final1= [each for each in string if each in consonants]
    print(len(final)) 
    print(len(final1))
    print(final) 
    print(final1)
vowels = "AeEeIiOoUu"
consonants="BbCcDdFfGgHhJjKkLlMmNnPpQqRrSsTtVvWwXxYyZz"
string='Siddiq'
Check(string, vowels,consonants)


2
4
['i', 'i']
['S', 'd', 'd', 'q']


Question 8:
Write a function that will take length and breadth for a rectangle and return perimeter and area of rectangle with 8% increment.


In [53]:
def areaRectangle(a, b): 
    h= (a * b)
    return h * (8/100) + h
      
def perimeterRectangle(a, b): 
    c = (2 * (a + b)) 
    return c * (8/100) + c
 
a = 5
b = 6 
print ("Area = ", areaRectangle(a, b) ) 
print ("Perimeter = ", perimeterRectangle(a, b))

Area =  32.4
Perimeter =  23.76


Question 9:
Write a function that can take the numbers in strings. From string find which number is even and which one is odd. Save them in two different lists and generate the result.


In [61]:
s="123456789"
e=0
o=0
even=[]
odd=[]
for i in s:
    if int(i)%2==0:
        e +=1
        even.append(i)
    else:
        o +=1
        odd.append(i)
print("Even numbers ",e)
print("odd numbers",o)
print("list of even",even)
print("list of odd",odd)

Even numbers  4
odd numbers 5
list of even ['2', '4', '6', '8']
list of odd ['1', '3', '5', '7', '9']


Question 10:
Write a function which will take the string from the user and return how many alphabets have been used in it and which alphabets are missing.


In [68]:
from string import ascii_lowercase

alphabet = set(ascii_lowercase)

def missing_letters(string):
    return alphabet - set(string.lower())

print(missing_letters("siddiq"))

{'f', 'u', 'e', 'w', 'h', 'k', 'b', 'n', 'm', 'c', 'a', 'o', 'x', 'r', 'v', 'y', 'z', 'j', 'l', 't', 'g', 'p'}


Question 11:
Write a function that will take verbs in words and return a list of verbs with continuous tense by adding (ing) at the end of each verb.


In [90]:
def verb(s):
    if s[-1] == 'e':
        s=s.replace(s[-1],"")
        print("verb:",s+"ing")
    else:
        print("verb:",s+"ing")
verb("live")

verb: living


Question 12:
Make a function which can take two radius of circles and find the areas of it and subtract smaller one from larger one and tell the remaining area of circle.


In [86]:
pi=3.142
def rArea(r,r1):
    a=pi*r**2
    a1=pi*r1**2
    if a > a1:
        left=(a1-a)
    else:
        left=(a-a1)
    return abs(left)
print(rArea(5,2))

65.982


Question 13:
Write a function that will take a string and calculate number of Upper case letters and lower case letters.


In [77]:
def up_low(s):      
    u = sum(1 for i in s if i.isupper())
    l = sum(1 for i in s if i.islower())
    print( "No. of Upper case characters : %s, No. of Lower case characters : %s" % (u,l))

up_low("My name is Siddiq?, Whats your name?")

No. of Upper case characters : 3, No. of Lower case characters : 24


Question 14:
Write a function which will take length and breadth of two rectangles. Subtract the smaller rectangle from the larger rectangle and return the area left behind.


In [83]:
def areaRectangle(a,b,aa,bb): 
    h= a*b
    h1=aa*bb
    if h > h1:
        left = (h1-h)
    else:
        left=(h-h1)
    return abs(left)
print(areaRectangle(5,6,4,3))
 

18


Question 15:
Create a function that can add the fractions in series such as 1 to 8 = 1/8+1/7+1/6+1/5 …… ½ and return the result in fraction not in decimal.


In [99]:
def fraction( numerator, denominator):
    min_val = min(numerator, denominator)
    for divisor in range(2, min_val+1):
        if (numerator % divisor == 0 and denominator % divisor == 0):
            return(fraction(numerator / divisor, denominator / divisor))
print('{}/{}'.format(4,6))

4/6


Question 16:
Write a function which will take height and base for a triangle and 


In [100]:
def area(a,b):
    h=(a*b)/2
    return h
area(2,3)

3.0

Question 17:
Write a function which will take a list of fruits names. The function will return how many alphabets are repetitive in the names of fruits and how many are unique letters.


In [3]:
mylist = ['apple', 'banana']


def count_letters(iterat):
    counted = {}
    for i in iterat:
        for l in i:
            if l in counted:
                counted[l] += 1
            else:
                counted[l] = 1
    return [[l for l in counted if counted[l] == 1], [l for l in counted if counted[l] > 1]]

count_letters(mylist)

[['l', 'e', 'b'], ['a', 'p', 'n']]

Question 18:
Write a function that can take square length and radius of circle. Find the area of both and subtract the smallest shape from largest one and return the remaining shape area.


In [6]:
pi=3.142
def l_r(d,r):
    sq=d**2
    cir=pi*r**2
    if sq>cir:
        remain=(sq-cir)
    else:
        remain=(cir-sq)
    return remain
l_r(2,4)

46.272